src/pages/SignIn/SignIn.jsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
+import React , { useState } from 'react';
+import axios from 'axios';
 import './SignIn.css';
+import { Link, useNavigate, useLocation } from 'react-router-dom';
+import { useUserAuth } from '../../context/userAuth-context';
+
 
 export default function SignIn() {
+  const navigate = useNavigate();
+   const [email , setEmail] = useState();
+   const [password , setPassword] = useState();
+   const [error , setError] = useState('');
+   const { setIsLogIn } = useUserAuth();  
+   const location = useLocation();
+   const from = location.state?.from?.pathname || "/" ;
+
+
+   const loginHandler = async(e) => {
+     e.preventDefault();
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email , password
+      });
+      if(response.status === 200){
+        localStorage.setItem("token", response.data.encodedToken);
+        setIsLogIn(true);
+        navigate(from , {replace:true});
+      } 
+    } catch (err) {
+        setError("The email you entered is not Registered. Please SignUp!");
```

Comment:
You cam also add a different error message for incorrect password. At present, even for a registered email, it shows 'Email not registered' error for wrong password.

Reply:
thanks for the suggestion will work on it.